### PR TITLE
Only enforce placement constraints for new or replaced pods

### PR DESCRIPTION
**Pull request by benclarkwood**
When placement constraints are updated, existing pods should not be moved to obey the constraints.

If an existing pod is restarted, it should not be moved to obey the constraint.

If a pod is added, it should obey the new constraint.

If a pod is replaced it should obey the new constraint.

### CODE REVIEW

File: frameworks/helloworld/tests/test_marathon_constraint.py [hunk 1]
```diff
@@ -5,15 +5,20 @@
...
 from tests.config import (
     PACKAGE_NAME
 )
 
 num_private_agents = len(shakedown.get_private_agents())
 
 
-def teardown_module(module):
-    install.uninstall(PACKAGE_NAME)
+# def teardown_module(module):
```
Comment (gabrielhartmann) [inline frameworks/helloworld/tests/test_marathon_constraint.py]: Commented code

File: frameworks/helloworld/tests/test_marathon_constraint.py [hunk 2]
```diff
@@ -127,3 +132,112 @@ def ensure_multiple_per_agent(hello, world):
...
+    out = cmd.run_cli('task {} --json'.format(task_name))
+    
+    for label in json.loads(out)[0]['labels']:
+        if label['key'] == 'offer_hostname':
+            return label['value']
+
+    raise Exception("offer_hostname label is not present!")
+
+
+# From Kafka
```
Comment (gabrielhartmann) [inline frameworks/helloworld/tests/test_marathon_constraint.py]: Unneeded comment

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/config/DefaultConfigurationUpdater.java [hunk 3]
```diff
@@ -271,8 +271,14 @@ private static boolean areDifferent(PodSpec podSpec1, PodSpec podSpec2) {
         }
 
         // Make counts equal, as only a difference in count should not effect an individual tasks.
-        podSpec1 = DefaultPodSpec.newBuilder(podSpec1).count(0).build();
-        podSpec2 = DefaultPodSpec.newBuilder(podSpec2).count(0).build();
+        podSpec1 = DefaultPodSpec.newBuilder(podSpec1).count(0).placementRule(null).build();
```
Comment (gabrielhartmann) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/config/DefaultConfigurationUpdater.java]: Comments are spread out between lines for no reason.  Make one large block comment.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java [hunk 4]
```diff
@@ -85,7 +85,10 @@ public OfferEvaluator(StateStore stateStore, OfferRequirementProvider offerRequi
     public List<OfferEvaluationStage> getEvaluationPipeline(OfferRequirement offerRequirement) {
         List<OfferEvaluationStage> evaluationPipeline = new ArrayList<>();
 
+        // OfferRequirement will not include placement rules,
+        // so it skips placement rules if it is a restart or transient failure
```
Comment (gabrielhartmann) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java]: This is commented elsewhere, and that functionality isn't implemented here.  Changes to this file should be removed.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/config/DefaultConfigurationUpdater.java [hunk 5]
```diff
@@ -271,8 +271,12 @@ private static boolean areDifferent(PodSpec podSpec1, PodSpec podSpec2) {
         }
 
         // Make counts equal, as only a difference in count should not effect an individual tasks.
```
Comment (gabrielhartmann) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/config/DefaultConfigurationUpdater.java]: This line is no longer needed.

**Comment by benclarkwood** [comment]
retest this please

**Pull request merged**